hw/src/lib.rs

```diff
@@ -128,84 +131,81 @@ impl From<libusb::Error> for Error {
 
 /// Hardware wallet management interface.
 pub struct HardwareWalletManager {
-	update_thread: Option<thread::JoinHandle<()>>,
+	active_threads: Vec<Option<thread::JoinHandle<()>>>,
```

Comment:
i'm curious about the motivation to have a separate thread for each HW wallet -- I am mildly worried about the overhead of waking these threads up fairly often to poll for exit and incurring context switches during regular import. 

The other downside to joining the threads manually is that bugs in libusb or drivers can cause Parity's exit to completely hang -- this is something we want to avoid because the `HardwareWalletManager` cleanup might be blocking e.g. rocksdb cleanup and the inevitable forced shutdown could break things.

This is a separate issue, but we have a `StopGuard` utility that currently wraps the `Arc<AtomicBool>` logic, with the idea that there is a single one shared by all the utilities in the program (with only the main thread capable of writing to it). I don't think we do this quite yet, but it would be nice to have the main thread signal stop and wait for a few seconds or for the ref-count to go to zero (whichever comes first).

Reply:
@rphmeier 
Great observations, the main motivation is to check whether `ÙSB Hotplug callbacks` are the reason of the crashes along with removing dummy for loops. By splitting it to two threads makes it possible the control callbacks in fine-grained manner instead of subscribing to all devices and the code becomes is to understand IMO. 

It's possible to do it with one thread and some additional checks though, but let's wait for the test on the Windows machine take it from there (I would be very surprised if it still fails... 🔨 )

Ok, didn't know that I just done similar to the previous implementation